mypy/subtypes.py

```diff
@@ -1374,6 +1377,10 @@ def visit_overloaded(self, left: Overloaded) -> bool:
     def visit_union_type(self, left: UnionType) -> bool:
         return all([self._is_proper_subtype(item, self.orig_right) for item in left.items])
 
+    def visit_type_guard_type(self, left: TypeGuardType) -> bool:
+        # TODO: What's the right thing to do here?
```

Comment:
I guess since we don't expect type guard types to be used for subtype checks, we can raise an error here as well?

Reply:
Thanks for reviewing! I'll make this one throw an error.

Reply:
Oops, that actually causes it to crash again. 
```
  File "/Users/jelle/py/mypy/mypy/checker.py", line 5232, in or_conditional_maps
    result[n1] = make_simplified_union([m1[n1], m2[n2]])
  File "/Users/jelle/py/mypy/mypy/typeops.py", line 370, in make_simplified_union
    if i != j and is_proper_subtype(tj, ti, keep_erased_types=keep_erased) and \
  File "/Users/jelle/py/mypy/mypy/subtypes.py", line 1178, in is_proper_subtype
    return _is_proper_subtype(left, right,
  File "/Users/jelle/py/mypy/mypy/subtypes.py", line 1199, in _is_proper_subtype
    return left.accept(ProperSubtypeVisitor(orig_right,
  File "/Users/jelle/py/mypy/mypy/types.py", line 290, in accept
    return visitor.visit_type_guard_type(self)
  File "/Users/jelle/py/mypy/mypy/subtypes.py", line 1381, in visit_type_guard_type
    raise RuntimeError("TypeGuard should not be used in subtype checks")
RuntimeError: TypeGuard should not be used in subtype checks
sphinx/ext/coverage.py:212: : note: use --pdb to drop into pdb
```
Going to write a proper implementation instead.